cmd/kubemark/hollow-node.go

```diff
@@ -173,23 +179,29 @@ func run(config *hollowNodeConfig) {
 		klog.Fatalf("Failed to create a ClientConfig: %v. Exiting.", err)
 	}
 
-	//TODO: arktos-scale-out: hollowNode should have the same args for Tenant servers as well
-	//      fixme when moving to this step running in perf test env
-	client := make([]clientset.Interface, 2)
-	clientFromConfig, err := clientset.NewForConfig(clientConfigs)
-	if err != nil {
-		klog.Fatalf("Failed to create a ClientSet: %v. Exiting.", err)
+	if len(config.ResourceServer) == 0 {
+		klog.Fatalf("Missing TenantServers. Exiting.")
 	}
 
+	numberTenantPartitions := len(config.ResourceServer)
+	client := make([]clientset.Interface, numberTenantPartitions)
 	for i := 0; i < len(client); i++ {
+		tenantCfg := restclient.CopyConfigs(clientConfigs)
+		for _, cfg := range tenantCfg.GetAllConfigs() {
+			cfg.Host = config.TenantServers[i]
+		}
+		clientFromConfig, err := clientset.NewForConfig(tenantCfg)
+		if err != nil {
+			klog.Fatalf("Failed to create a ClientSet: %v. Exiting.", err)
+		}
 		client[i] = clientFromConfig
 	}
 
 	if config.Morph == "kubelet" {
 		// Start APIServerConfigManager
-		// TODO: Arktos-scale-out: evaluate this function and see if this needs to be done on all tenant partitions
-		//
-		go datapartition.StartAPIServerConfigManagerAndInformerFactory(client[0], wait.NeverStop)
+		for _, clt := range client {
+			go datapartition.StartAPIServerConfigManagerAndInformerFactory(clt, wait.NeverStop)
```

Comment:
man this is a long name

Reply:
yes. some style. please not this is existing code so i won't intent to change it at all :-)